10_predictive-modeling-review.qmd

```diff
@@ -0,0 +1,1045 @@
+
+First, finalize the workflow with the "best" hyperparameters from the tuned results. 
+
+```{r}
+final_wf <- 
+  knn_workflow |> 
+  finalize_workflow(select_best(knn_res))
+
+```
+
+Second, use `last_fit()` to fit the model on all of the data. 
+
+```{r}
+final_fit <- 
+  final_wf |>
+  last_fit(data1_split) 
+
+```
+
+::: {.callout-note}
+
+`select_best()` takes a narrow view of model selection and picks the model with the lowest error rate. 
+
+It is important to consider other elements when picking a "best". Other considerations include parsimony, cost, and equity. 
+
+:::
+
+#### 7. Evaluate the Final Model
+
+The final fit object contains the out-of-sample error metric from the testing data. This metric is our best estimate of model performance on unseen data. 
+
+In this case, the testing data error is slightly higher than the training data error, which makes sense. 
+
+```{r}
+final_fit |>
+  collect_metrics()
+
+```
+
+#### 8. Optional: Implement the Final Model
+
+`final_fit` and `pedict()` can be used to apply the model to new, unseen data. 
+
+Only implement the model if it achieves the objectives of the final model. 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+### Example: Regression Trees
+
+We'll next work with a subset of data from the `Chicago` data set from `library(tidymodels)`. 
+
+```{r}
+chicago_small <- Chicago |>
+  select(
+    ridership,
+    Clark_Lake,
+    Quincy_Wells,
+    Irving_Park,
+    Monroe,
+    Polk,
+    temp,
+    percip,
+    Bulls_Home,
+    Bears_Home,
+    WhiteSox_Home,
+    Cubs_Home,
+    date
+  ) |>
+  mutate(weekend = wday(date, label = TRUE) %in% c("Sat", "Sun")) |>
+  glimpse()
+
+```
+
+#### 0. Problem Formulation
+
+The variable `ridership` is the outcome of interest. It measures ridership at the Clark/Lake L station in Chicago. The variables `Clark_Lake`, `Quincy_Wells`, `Irving_Park`, `Monroe`, and `Polk` measure ridership at several stations *14 days before* the `ridership` variable. 
+
+The objective is predict `ridership` to better allocate staff and resources to the Clark/Lake L station. 
+
+#### 1. Split data into training and testing data
+
+```{r}
+set.seed(20231106)
+
+# create a split object
+chicago_split <- initial_split(data = chicago_small, prop = 0.8)
+
+# create the training and testing data
+chicago_train <- training(x = chicago_split)
+chicago_test  <- testing(x = chicago_split)
+
+```
+
+#### 2. Exploratory Data Analysis
+
+[This chapter](https://bookdown.org/max/FES/visualizations-for-numeric-data-exploring-train-ridership-data.html) in "Feature and Target Engineering" contains EDA for the data set of interest.
+
+#### 3. Set up Resampling for Model Selection
+
+We use 10-fold cross validation for hyperparameter tuning and model selection. The training data are larger in the previous example, so we skip repeats. 
+
+```{r}
+chicago_folds <- vfold_cv(data = chicago_train, v = 10)
+
+```
+
+#### 4. Create Candidate Models
+
+We will use regression trees for this example. 
```

Comment:
Add a sentence saying:

For feature and target engineering, we will simply remove the date column

Reply:
Added.